### ISSUE TITLE
[C#] Implement new Concept Exercise: method-overloading

### ISSUE THREAD

**Issue by ErikSchierboom** [exercism/v3#1046]
This issue describes how to implement the `method-overloading` concept exercise for the C# track.

## Getting started

**Please please please read the docs before starting.** Posting PRs without reading these docs will be a lot more frustrating for you during the review cycle, and exhaust Exercism's maintainers' time. So, before diving into the implementation, please read up on the following documents:

- [The features of v3](https://github.com/exercism/v3/blob/master/docs/concept-exercises.md).
- [Rationale for v3](https://github.com/exercism/v3/blob/master/docs/rationale-for-v3.md).
- [What are concept exercise and how they are structured?](https://github.com/exercism/v3/blob/master/docs/features-of-v3.md)

Please also watch the following video:

- [The Anatomy of a Concept Exercise](https://www.youtube.com/watch?v=gkbBqd7hPrA).

## Goal

The goal of this exercise is to teach the student more advanced aspects of the Concept of Method Overloading in C#.

## Learning objectives

- Know what method overloading is.
- Know how to define overloaded methods.
- Know the limitations of method overloading.
- Know how to define optional parameters.
- Know how to pass named arguments.

## Out of scope

- Overload resolution.

## Concepts

This Concepts Exercise's Concepts are:

- `method-overloading`: know what method overloading is; know how to define overloaded methods; know the limitations of method overloading.
- `optional-parameters`: know how to define optional parameters.
- `named-arguments`: know how to use named arguments.

## Prequisites

This Concept Exercise's prerequisites Concepts are:

- `classes`: know how to define methods on classes.

Any data types used in this exercise (e.g. `strings`) should also be added as prerequisites.

### Hints

- [Member overloading][member-overloading]: how to overload members.
- [Named and optional parameters][named-and-optional-parameters]: what are named and optional parameters and how to use them.

### After

- [Member overloading][member-overloading]: how to overload members.
- [Named and optional parameters][named-and-optional-parameters]: what are named and optional parameters and how to use them.

## Representer

This exercise does not require any specific representation logic to be added to the [representer][representer].

## Analyzer

This exercise does not require any specific analyzer logic to be added to the [analyzer][analyzer].

## Implementing

To implement this exercise, please [follow these instructions](https://github.com/exercism/v3/blob/master/languages/csharp/reference/implementing-a-concept-exercise.md).

## Help

If you have any questions while implementing the exercise, please post the questions as comments in this issue.

[how-to-implement-a-concept-exercise]: https://github.com/exercism/v3/blob/master/docs/maintainers/generic-how-to-implement-a-concept-exercise.md
[implemented-exercises]: https://github.com/exercism/v3/tree/master/languages/csharp/exercises/concept/README.md#implemented-exercises
[reference]: https://github.com/exercism/v3/blob/master/languages/csharp/reference/README.md#reference-docs
[reference-doc]: https://github.com/exercism/v3/blob/master/reference/concepts/method-overloading.md
[reference-example]: https://github.com/exercism/v3/blob/master/reference/types/string.md#implementations
[analyzer]: https://github.com/exercism/csharp-analyzer
[representer]: https://github.com/exercism/csharp-representer
[exercise-example]: https://github.com/exercism/v3/tree/master/languages/csharp/exercises/concept/numbers-floating-point
[design-example]: https://github.com/exercism/v3/blob/master/languages/csharp/exercises/concept/numbers/.meta/design.md
[config.json-example]: https://github.com/exercism/v3/blob/master/languages/csharp/exercises/concept/numbers/.meta/config.json
[concept-exercises]: https://github.com/exercism/v3/blob/master/docs/concept-exercises.md
[member-overloading]: https://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/member-overloading
[named-and-optional-parameters]: https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/classes-and-structs/named-and-optional-arguments

**Comment by batibot323** [comment]
The assignee is the one that has to come up with the problem, right? The problem that will serve as opportunity to teach and apply this Concept?

**Comment by ErikSchierboom** [comment]
@batibot323 Well, usually: yes. Unless you find someone else to come up with a story. Regarding this issue: I'm currently restructuring them, so could you perhaps hold off on this issue for a bit?

If you'd like to jump into an issue, the [booleans exercise](https://github.com/exercism/v3/issues/1004) is perfect, as it has already been implemented in F#: https://github.com/exercism/v3/tree/master/languages/fsharp/exercises/concept/booleans You can then use that as the source for porting it to C#.

**Comment by ErikSchierboom** [comment]
I'm working on this.
